meson.build

```
@@ -0,0 +1,27 @@
+project('crypto', 'c', version: '0.0.1', license: 'MIT')
+cc = meson.get_compiler('c')
+
+
+
+gmpDep = dependency('gmp', required: get_option('USE_GMP'))
+
+if gmpDep.found() and get_option('USE_GMP')
+  add_global_arguments('-DUSE_GMP=true', language: 'c')
```

```suggestion
  add_project_arguments('-DUSE_GMP=true', language: 'c')
```

Exactly the same, unless you use subprojects, in which case add_global_arguments will either error out or define this argument in the subproject, which you probably didn't expect.